pkg/consensus/consensus.go

```diff
@@ -677,9 +677,22 @@ func (s *service) getValidators(txes ...block.Transaction) []crypto.PublicKey {
 		err   error
 	)
 	if txes == nil {
+		// getValidators with empty args is used by dbft to fill the list of
+		// block's validators, thus should return validators from the current
+		// epoch without recalculation.
 		pKeys, err = s.Chain.GetNextBlockValidators()
 	} else {
-		pKeys = s.Chain.ComputeNextBlockValidators()
+		// getValidators with non-empty args is used by dbft to fill block's
+		// NextConsensus field, thus should return proper value for NextConsensus.
+		cfg := s.Chain.GetConfig().ProtocolConfiguration
+		if cfg.ShouldUpdateCommitteeAt(s.dbft.Context.BlockIndex) {
```

Comment:
Then here you won't need to think about indexes, just request `NextNext` list.

Reply:
Fixed. Although it's just an improvement, and there's still a bug somewhere in the code, I'm investigating it.